ls1intum/Artemis
# `Exam mode`: Fix live monitoring issues on Chrome and sorting of import dialog

### Checklist
#### General


#### Server
#### Client

### Motivation and Context



Fixes issue no. 17, and no.22 (Closes #5716) found during [exam mode testing](https://confluence.ase.in.tum.de/pages/viewpage.action?pageId=138610767).

### Description

Bug No 17: Clicking on the sort arrows in the import modal for an exam results in an error.
Problem: 
- The exam mode ordering was broken because the sorted column was `exam.testExam` instead of just `testExam`.
- The course title ordering was broken as DISTINCT was used. 

Bug No 22: Exam Monitoring page broken on chrome because of thousand of errors.
Problem: Same Error as described previously here;
https://github.com/ls1intum/Artemis/pull/4479
```
You might notice that the appearance of these bar charts differ a bit from other implementations with ngx-charts. The edges of the bars are not rounded here. 

This is due to an error output I was facing while testing this implementation locally (to be more precise, this error only occurs when the "show (un-)rated results" button is triggered). 

d3 seems to have issues when it comes to changing bars representing a non-zero value to zero. 

It is a bit weird because this error occurred for all charts except the point distribution chart (where the value 0 occurs as well). I did some research and found 

https://github.com/swimlane/ngx-charts/issues/498. There, disabling the edge rounding was proposed as workaround, which worked for me.
```
Solution: Deactivated rounding of bar charts in exam monitoring and all other instances of bar charts I found in the codebase

### Steps for Testing
Bug 17: Try to import an exam and sort by exam mode property. This should work and sorting by course title should be deactivated
Bug 22: Test that exam monitoring page is not broken on Chrome (open console)

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
- No UI changes

### Screenshots
- No UI changes
